src/tribler/core/components/ipv8/eva_protocol.py

```diff
@@ -144,32 +144,66 @@ class TransferResult:
     data: bytes
     nonce: int
 
+    def __str__(self):
+        return f'TransferResult(peer={self.peer}, info: {self.info}, data hash: {hash(self.data)}, nonce={self.nonce})'
+
+class TransferWindow:
+    def __init__(self, start: int, size: int):
+        self.data: List[Optional[bytes]] = [None for _ in range(size)]
+
+        self.size = size
+        self.start = start
+        self.processed: int = 0
+        self.last_window: bool = False
+
+    def add(self, index: int, block: bytes):
+        if self.data[index] is not None:
+            return
+        self.data[index] = block
+        self.processed += 1
+
+    def finished(self) -> bool:
```

Comment:
It is better to rename it to `is_finished`

Reply:
renamed